client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -1509,8 +1558,18 @@ public CompletableFuture<InsertResponse> insert(String tableName,
 
         settings.setOption(ClientConfigProperties.INPUT_OUTPUT_FORMAT.getKey(), format.name());
         final InsertSettings finalSettings = settings;
-        final String sqlStmt = "INSERT INTO " + tableName + " FORMAT " + format.name();
-        finalSettings.serverSetting(ClickHouseHttpProto.QPARAM_QUERY_STMT, sqlStmt);
+
```

Comment:
Should it have a condition if empty `"INSERT INTO " + tableName + " FORMAT " + format.name();` else do the loop